src/libcsg/modules/io/pdbwriter.cc

```diff
@@ -73,7 +73,7 @@ void PDBWriter::WriteBox(const Eigen::Matrix3d &box) {
   _out << boxfrmt % a % b % c % alpha % beta % gamma;
 }
 
-void PDBWriter::writeSymmetry(Bead *bead) {
+void PDBWriter::writeSymmetry(const std::unique_ptr<Bead> &bead) {
```

Comment:
I would prefer just a `const Bead& bead`

Reply:
Ya but that does not work well with the vector of unique pointers stored in the topology class.

Reply:
You would have to do something special to get the begin and end iterator methods to work like that. If you can figure out how I will implement it though.

Reply:
you can just call `writeSymmetry(*bead)` then and dereference the pointer.

Reply:
which call of this function do you think is problematic?